src/app/containers/app-header/app-header.component.ts

```diff
@@ -128,17 +121,22 @@ export class AppHeaderComponent implements OnInit, OnDestroy {
     }
   }
 
-  public getApplicationThemeForUser(): Observable<ApplicationTheme> {
-    try {
-      return this.getUsersTheme(this.defaultTheme);
-    } catch (error) {
-      return this.logErrorAndReturnDefaultTheme(error);
-    }
-  }
-
-  public logErrorAndReturnDefaultTheme(error): Observable<ApplicationTheme> {
-    this.loggerService.error(error);
-    return of(this.defaultTheme);
+  public setApplicationThemeForUser(): void {
+    const availableThemes = environment.themes;
+    const userRolesSet = new Set(this.userRoles);
+    const userTheme = Object.keys(availableThemes).find((theme) =>
+      theme !== 'default' && theme.split('|').some((role) => userRolesSet.has(role))
```

Comment:
Can you make the role matching for themes a regex please. What you've done works, but isn't functionally equivalent to what we had. You should also change default to '.+'
Then you could do something like 
const userTheme = Object.keys(availableThemes.some((themeRegex) => userRolesSet.some((role) => role.matches(themeRegex)))